### PR TITLE
Update playground

### PR DESCRIPTION
**Pull request by JeffGuKang**
Update loop section in Playground.
Arrange source.

Next will be updated `error` section.

### CODE REVIEW

**Comment by JeffGuKang** [comment]
Next will be `error` section.

**Comment by ldiqual** [comment]
@JeffGuKang Neat! Would you mind discarding your changes in `contents.xcplayground` and `contents.xcworkspacedata`? Thank you!

**Comment by JeffGuKang** [comment]
@ldiqual I updated following your suggestions. :)

**Comment by ldiqual** [comment]
@JeffGuKang Still shows up in the diff: https://github.com/SwiftyJSON/SwiftyJSON/pull/739/files#diff-baa084bbb5cafa8d9e3a530988a38ca4

**Comment by JeffGuKang** [comment]
I think github recommend about including new line at the end of file. What is better?

**Comment by ldiqual** [comment]
@JeffGuKang We should probably fix this as part of a PR that is more related to it, like the Xcode 8.2 upgrade or something like that. It's more about keeping the history clean and avoiding conflicts, I don't mind too much about the change itself but it can be misleading for git blaming.

**Comment by JeffGuKang** [comment]
@ldiqual You are right. I rewinded the file using copy from some previous commit having newline warning.

File: Example/Playground.playground/Contents.swift [hunk 1]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+// With a custom way
 let keys: [JSONSubscriptType] = ["users", 1, "info", "name"]
-name = jsonArray[keys].string
+name = json[keys].string
+
+// Just the same
+name = json["users"][1]["info"]["name"].string
+
+// Alternatively
+name = json["users",1,"info","name"].string
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Checked

File: Example/Playground.playground/Contents.swift [hunk 2]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+
+/*:
+ ### Loop
+ */
+// If json is .Dictionary
+for (key,subJson):(String, JSON) in json {
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Checked

File: Example/Playground.playground/Contents.swift [hunk 3]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+	//Do something you want
+//	print(subJson)
+}
+
+/*The first element is always a String, even if the JSON is an Array*/
+//If json is .Array
+//The `index` is 0..<json.count's string value
+for (index,subJson):(String, JSON) in json["array"] {
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Checked

File: Example/Playground.playground/Contents.swift [hunk 4]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+	//Do something you want
+	print("\(index): \(subJson)")
+}
+
+/*:
+ ### Error
+
+SwiftyJSON 4.x
+
+SwiftyJSON 4.x introduces an enum type called `SwiftyJSONError`, which includes `unsupportedType`, `indexOutOfBounds`, `elementTooDeep`, `wrongType`, `notExist` and `invalidJSON`, at the same time, `ErrorDomain` are being replaced by `SwiftyJSONError.errorDomain`. Note: Those old error types are deprecated in SwiftyJSON 4.x and will be removed in the future release.
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Can be ignored

File: Example/Playground.playground/Contents.swift [hunk 5]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+*/
+
+let errorJson = JSON(["name", "age"])
+if let name = errorJson[999].string {
+	//Do something you want
+} else {
+	print(errorJson[999].error!) // "Array[999] is out of bounds"
+}
+
+let errorJson2 = JSON(["name":"Jack", "age": 25])
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Checked

File: Example/Playground.playground/Contents.swift [hunk 6]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+
+// Example json
+let jsonOG: JSON = JSON([
+	"id": 987654,
+	"user": [
+		"favourites_count": 8,
+		"name": "jack",
+		"email": "jack@gmail.com",
+		"is_translator": true
+	],
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Checked

File: Example/Playground.playground/Contents.swift [hunk 7]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+	"list": [
+		["number": 1],
+		["number": 2],
+		["number": 3]
+	],
+	"user": [
+		"favourites_count": 8,
+		"email": "jack@gmail.com",
+		"is_translator": true
+	],
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Checked

File: Example/Playground.playground/Contents.swift [hunk 8]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+//If not a Number or nil, return 0
+let idNOG: Int = jsonOG["id"].intValue
+print(idNOG)
+
+//If not a String or nil, return ""
+let nameNOG: String = jsonNOG["name"].stringValue
+print(nameNOG)
+
+//If not an Array or nil, return []
+let listNOG: Array<JSON> = jsonNOG["list"].arrayValue
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Checked

File: Example/Playground.playground/Contents.swift [hunk 9]
```diff
@@ -66,33 +66,174 @@ var jsonArray: JSON = [
...
+print(listNOG)
+
+//If not a Dictionary or nil, return [:]
+let userNOG: Dictionary<String, JSON> = jsonNOG["user"].dictionaryValue
```
Comment (JeffGuKang) [inline Example/Playground.playground/Contents.swift]: Checked

**Comment by JeffGuKang** [comment]
Moved to #905

**Pull request closed without merging**